gateway-server/src/main/java/org/apache/knox/gateway/services/factory/AliasServiceFactory.java

```diff
@@ -44,22 +41,17 @@ public Service create(GatewayServices gatewayServices, ServiceType serviceType,
       final AliasService defaultAliasService = new DefaultAliasService();
       ((DefaultAliasService) defaultAliasService).setMasterService(getMasterService(gatewayServices));
       ((DefaultAliasService) defaultAliasService).setKeystoreService(getKeystoreService(gatewayServices));
-      defaultAliasService.init(gatewayConfig, options); //invoking init on DefaultAliasService twice is ok (in case implementation is set to 'default')
-      switch (implementation) {
-      case DEFAULT_IMPLEMENTATION_NAME:
-      case "":
+      defaultAliasService.init(gatewayConfig, options); // invoking init on DefaultAliasService twice is ok (in case implementation is set to 'default')
+
+      if (matchesImplementation(implementation, DefaultAliasService.class, true)) {
```

Comment:
Why is the DefaultAliasService always created and initialized?
Could this not be done only when there is no implementation, or if the specified implementation is the default one?

Reply:
If you check out the rest of the implementations, they all use the `defaultAliasService` instance for their purpose. It was the same way (as a local alias service)